pkg/system/phase4_configuring.go

```diff
@@ -541,16 +545,109 @@ func (r *Reconciler) awaitEndpointDeploymentPods() error {
 func (r *Reconciler) ReconcileHPAEndpoint() error {
 	// Wait for the the endpoint deployment to become ready
 	// only if HPA was not created yet
+	log := r.Logger.WithField("func", "ReconcileHPAEndpoint")
 	if r.HPAEndpoint.UID == "" {
 		if err := r.awaitEndpointDeploymentPods(); err != nil {
 			return err
 		}
 	}
+	var err error
+	prometheusList := deployedPrometheus(log)
+	switch r.NooBaa.Spec.AutoscalingType {
+	case "keda":
+		err = r.autoscaleKeda(prometheusList)
+	case "hpav2":
+		//TODO
+		log.Infof("HPAV2 implementation pending")
+	default:
+		if err := r.ReconcileObject(r.HPAEndpoint, r.SetDesiredHPAEndpoint); err != nil {
+			return err
+		}
+	}
+	if err != nil {
```

Comment:
I think there shouldn't be a fallback in case of an error in deploying the autoscaler. this error should be returned back and treated as any other temp error. the operator will retry in the next reconcile

Reply:
@dannyzaken if we return the error back to the main reconcile method `ReconcilePhaseConfiguring`, then it will skip the rest of the flow in that reconcile method, is that Ok?